video/out/opengl/utils.c

```diff
@@ -1019,3 +1019,77 @@ void gl_sc_gen_shader_and_reset(struct gl_shader_cache *sc)
 
     gl_sc_reset(sc);
 }
+
+void gl_timer_init(struct gl_timer *timer, GL *gl)
+{
+    *timer = (struct gl_timer){ .gl = gl };
+
+    if (gl->GenQueries)
+        gl->GenQueries(QUERY_OBJECT_NUM, timer->query);
+}
+
+void gl_timer_uninit(struct gl_timer *timer)
+{
+    GL *gl = timer->gl;
+    if (gl->DeleteQueries)
+        gl->DeleteQueries(QUERY_OBJECT_NUM, timer->query);
```

Comment:
uninit should probably be idempotent and also work on an not initialized struct (i.e. check for gl and reset the query IDs).